File: 프로그래머스/3/12927. 야근 지수/야근 지수.py
import heapq
def solution(n, works):
    answer = 0
    heap = []
    heapq.heapify(heap)
    for i in works :
        heapq.heappush(heap,-i)
    for i in range(n):
        if heap : 
            a = heapq.heappop(heap)
            a += 1
            if a != 0 :
                heapq.heappush(heap,a)
    #print(heap)
    for i in heap :
        answer += (-i) ** 2
    
    return answer

File: 프로그래머스/3/12927. 야근 지수/야근 지수.java
import java.util.*;
import java.io.*;

class Solution {
    public long solution(int n, int[] works) {
        long answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> b - a);
        for(int work : works)
            pq.add(work);
        while(!pq.isEmpty() && n > 0){
            int num = pq.poll();
            num--;
            n--;
            if(num > 0)
                pq.add(num);
        }
        
        while(!pq.isEmpty()){
            int num = pq.poll();
            answer += (num * num);
        }
        
        return answer;
    }
}